src/pe/resource.rs

```diff
@@ -0,0 +1,1942 @@
+pub const VFT2_DRV_SYSTEM: u32 = 0x00000007;
+/// [`VsFixedFileInfo::file_subtype`]: Indicates an installable driver subtype.
+pub const VFT2_DRV_INSTALLABLE: u32 = 0x00000008;
+/// [`VsFixedFileInfo::file_subtype`]: Indicates a sound driver subtype.
+pub const VFT2_DRV_SOUND: u32 = 0x00000009;
+/// [`VsFixedFileInfo::file_subtype`]: Indicates a communication driver subtype.
+pub const VFT2_DRV_COMM: u32 = 0x0000000A;
+/// [`VsFixedFileInfo::file_subtype`]: Indicates an input method driver subtype.
+pub const VFT2_DRV_INPUTMETHOD: u32 = 0x0000000B;
+/// [`VsFixedFileInfo::file_subtype`]: Indicates a versioned printer driver subtype.
+pub const VFT2_DRV_VERSIONED_PRINTER: u32 = 0x0000000C;
+
+// VS_VERSION.dwFileSubtype for VFT_WINDOWS_FONT
+
+/// [`VsFixedFileInfo::file_subtype`]: Indicates a raster font subtype.
+pub const VFT2_FONT_RASTER: u32 = 0x00000001;
+/// [`VsFixedFileInfo::file_subtype`]: Indicates a vector font subtype.
+pub const VFT2_FONT_VECTOR: u32 = 0x00000002;
+/// [`VsFixedFileInfo::file_subtype`]: Indicates a TrueType font subtype.
+pub const VFT2_FONT_TRUETYPE: u32 = 0x00000003;
+
+/// Iterator over [`ResourceString`]
+#[derive(Debug, Copy, Clone)]
+pub struct ResourceStringIterator<'a> {
+    /// The raw data must be scoped to the [`ResourceDataEntry`]
+    data: &'a [u8],
+}
+
+impl<'a> Iterator for ResourceStringIterator<'a> {
+    type Item = error::Result<ResourceString<'a>>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.data.is_empty() {
+            return None;
+        }
+
+        let mut offset = 0;
+        Some(match ResourceString::parse(self.data, &mut offset) {
+            Ok(next) => {
+                debug!(
+                    "Parsed next resource string as size {:#x}: {:#x?}",
+                    offset, next?
+                );
+                self.data = &self.data[offset..];
+                Ok(next?)
+            }
+            Err(error) => {
+                self.data = &[];
+                Err(error.into())
+            }
+        })
+    }
+}
+
+/// Represents a resource string entry.
+#[derive(Copy, Clone, PartialEq)]
+pub struct ResourceString<'a> {
+    /// The length, in bytes, of this [`ResourceString`] structure.
+    pub len: u16,
+    /// The size, in words, of the [`ResourceString::value`].
+    ///
+    /// - When [`ResourceString::type`] indicates string data: multiply this field with [`SIZE_OF_WCHAR`] that should be the actual size of [`ResourceString::value`] with null-terminator.
+    /// - Othereise, treat as-is.
+    pub value_len: u16,
+    /// The type of [`ResourceString::value`] in the version resource.
+    ///
+    /// This member is `1` if the version resource contains text data;
+    /// and `0` if the version resource contains binary data, otherwise sometimes an invalid value.
+    pub r#type: u16,
+    /// An arbitrary null-terminated utf-16 unicode string.
+    pub key: Utf16String<'a>,
```

Comment:
this is the only place Utf16String is used (publicly, afaics); i would really prefer to keep this pub(crate) for now since once this datatype escapes out into pub, it's very hard to change. Since the other uses have methods to just get the key_string and value_string out, I think it's ok to remove pub from this for now until we've really nailed down the Utf16String naming and representation, etc.